plugins/modules/ec2_vpc_nat_gateway.py

```diff
@@ -491,199 +390,188 @@ def gateway_in_subnet_exists(client, subnet_id, allocation_id=None,
-
-    Kwargs:
-        check_mode (bool): if set to true, do not run anything and
-            falsify the results.
+        module: AnsibleAWSModule class instance
 
     Basic Usage:
         >>> client = boto3.client('ec2')
-        >>> allocate_eip_address(client)
+        >>> module = AnsibleAWSModule(...)
+        >>> allocate_eip_address(client, module)
         True
 
     Returns:
         Tuple (bool, str)
     """
-    ip_allocated = False
+
     new_eip = None
-    err_msg = ''
+    msg = ''
+
     params = {
         'Domain': 'vpc',
     }
-    try:
-        if check_mode:
-            ip_allocated = True
-            random_numbers = (
-                ''.join(str(x) for x in random.sample(range(0, 9), 7))
-            )
-            new_eip = 'eipalloc-{0}'.format(random_numbers)
-        else:
-            new_eip = client.allocate_address(aws_retry=True, **params)['AllocationId']
-            ip_allocated = True
-        err_msg = 'eipalloc id {0} created'.format(new_eip)
 
-    except botocore.exceptions.ClientError as e:
-        err_msg = str(e)
+    if module.check_mode:
+        ip_allocated = True
+        new_eip = None
+        return ip_allocated, msg, new_eip
+
+    try:
+        new_eip = client.allocate_address(aws_retry=True, **params)['AllocationId']
+        ip_allocated = True
+        msg = 'eipalloc id {0} created'.format(new_eip)
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e)
 
-    return ip_allocated, err_msg, new_eip
+    return ip_allocated, msg, new_eip
 
 
-def release_address(client, allocation_id, check_mode=False):
+def release_address(client, module, allocation_id):
     """Release an EIP from your EIP Pool
     Args:
         client (botocore.client.EC2): Boto3 client
+        module: AnsibleAWSModule class instance
         allocation_id (str): The eip Amazon identifier.
 
-    Kwargs:
-        check_mode (bool): if set to true, do not run anything and
-            falsify the results.
-
     Basic Usage:
         >>> client = boto3.client('ec2')
+        >>> module = AnsibleAWSModule(...)
         >>> allocation_id = "eipalloc-123456"
-        >>> release_address(client, allocation_id)
+        >>> release_address(client, module, allocation_id)
         True
 
     Returns:
         Boolean, string
     """
-    err_msg = ''
-    if check_mode:
+
+    msg = ''
+
+    if module.check_mode:
         return True, ''
 
     ip_released = False
+
     try:
         client.describe_addresses(aws_retry=True, AllocationIds=[allocation_id])
-    except botocore.exceptions.ClientError as e:
+    except is_boto3_error_code('InvalidAllocationID.NotFound') as e:
         # IP address likely already released
         # Happens with gateway in 'deleted' state that
         # still lists associations
-        return True, str(e)
+        return True, e
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
```

Comment:
```suggestion
    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:  # pylint: disable=duplicate-except
```

Reply:
Thanks. I forget to commit it. Could you kindly have a look as well at this https://github.com/ansible-collections/community.aws/pull/436 (when you will have some time)?